### PR TITLE
Rewrite `Image` archetype

### PR DESCRIPTION
**Pull request by emilk**
### What
* Part of https://github.com/rerun-io/rerun/issues/6844
* Closes https://github.com/rerun-io/rerun/issues/6386

### Breaking changes
* `Image(…).compress(…)` is now `ImageEncoded.compres(…)`
* `ImageEncoded` has very different constructor compared to 0.17, and only supports chroma-downsampled images

### TODO
* [x] Fix how enums are handled in Python (I don't understand it). In particular, this code is broken when given a string as a color model: https://github.com/rerun-io/rerun/blob/8974966ef17c71574df04a77e9ffee0125b32063/rerun_py/rerun_sdk/rerun/archetypes/image_encoded_ext.py#L150-L157
* [x] Fix this test: https://github.com/rerun-io/rerun/blob/4f12d7f6f47a15e0583c5de9e2883840b571e21e/rerun_py/tests/unit/test_image.py#L16-L45
* [x] Test `detect_and_track_objects`
* [x] Test a bunch of other examples
* ~~[ ] Log a warning if `color_model` is missing, so that we can in the future migrate to having BGR be the default~~
  * BGR should never be the default. This has haunted opencv for over a decade. We need not inherit this trauma.
* ~~[ ] Consider re-implementing `image.compress(…)` (with a forward call to `ImageEncoded.compress`~~
  * TODO: I don't understand why this was moved in the first place. We should move it back, but it can happen in a follow-on PR.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6942?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6942?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide:
  * https://github.com/rerun-io/rerun/issues/7031

- [PR Build Summary](https://build.rerun.io/pr/6942)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: rerun_py/rerun_sdk/rerun/datatypes/tensor_buffer.py [hunk 1]
```diff
@@ -81,66 +81,6 @@ class TensorBuffer(TensorBufferExt):
...
-
-        First comes entire image in Y, followed by interleaved lines ordered as U0, V0, U1, V1, etc.
-
-    * YUY2 (npt.NDArray[np.uint8]):
-        YUY2, also known as YUYV is a YUV 4:2:2 chroma downsampled format with 8 bits per channel.
-
-        The order of the channels is Y0, U0, Y1, V0.
-    """
-
-    kind: Literal["u8", "u16", "u32", "u64", "i8", "i16", "i32", "i64", "f16", "f32", "f64", "nv12", "yuy2"] = field(
```
Comment (emilk) [inline rerun_py/rerun_sdk/rerun/datatypes/tensor_buffer.py:96]: Where did this go??? O.o

**Comment by emilk** [comment]
@rerun-bot full-check

File: crates/build/re_types_builder/src/codegen/cpp/mod.rs [hunk 2]
```diff
@@ -336,17 +336,21 @@ fn hpp_type_extensions(
...
     }
 
-    let comment = quote_comment(&format!(
-        "Extensions to generated type defined in '{}'",
+    let start_comment = quote_comment(&format!(
+        "START of extensions from {}:",
+        extension_file.file_name().unwrap()
+    ));
+    let end_comment = quote_comment(&format!(
+        "END of extensions from {}, start of generated code:",
```
Comment (Wumpf) [inline crates/build/re_types_builder/src/codegen/cpp/mod.rs:344]: srsly why did this have to go into _this_ PR -.-

File: docs/content/reference/migration/migration-0-18.md [hunk 3]
```diff
@@ -12,13 +12,36 @@ The `DepthImage` and `SegmentationImage` archetypes used to be encoded as a tens
...
+* Python: `NV12/YUY2` are now logged with the new `Image`:
+
+
+```py
+rr.log(
+    "my_image",
+    rr.Image(
+        bytes=.…,
+        width=.…,
+        height=.…,
```
Comment (Wumpf) [inline docs/content/reference/migration/migration-0-18.md:39]:
```suggestion
        bytes=…,
        width=…,
        height=…,
```

**Review by Wumpf — APPROVED** [review]

File: crates/build/re_types_builder/src/codegen/cpp/mod.rs (hunk 2 above)
Comment (jleibs) [reply crates/build/re_types_builder/src/codegen/cpp/mod.rs:344]: Yeah -- if Emil was still here I would have asked to split that out.

**Pull request merged**